internal/index/manager/manager.go

```diff
@@ -821,6 +830,32 @@ func (mgr *Manager) getIndexesCopy(start int) ([]*index.Reader, indexReleaser) {
 	return indexes, mgr.lock(indexes)
 }
 
+func (mgr *Manager) SetClientConfig(config ClientConfig) ClientConfig {
+	c := make(chan ClientConfig)
+	mgr.jobs <- func() {
+		mgr.clientConfig = config
+
+		mgr.event(Event{
+			Type:   "configUpdated",
+			Config: config,
+		})
+		close(c)
+	}
+
+	res := <-c
```

Comment:
You never send anything to the channel and don't use the return value of `SetClientConfig` so this can just be a `make(chan struct{})` and a `<-c` after adding the job to wait for the channel to be closed instead of returning something.